File: home_node.py
'''
Object to represent smart home node
'''
from bs4 import BeautifulSoup
import requests

class home_node:
	def __init__(self, ip, name):
		self.ip = ip
		self.name = name
		self.num_relays = 4

	#Set relay based on old arduino code
	def set_relay_old(self, relay_num):
		try:
			requests.get('http://{}/relay{}'.format(self.ip, relay_num))

		except requests.exceptions.RequestException as e:	
			print (e)

	''''''
	def set_relay(self, relay_num, state):
		try:
			requests.get('http://{}/re{}_{}'.format(self.ip, relay_num, state))

		except requests.exceptions.RequestException as e:	
			print (e)

	# def set_all(self, state):
	# 	if state is 'on':
	# 		set_relay()


	def get_temp(self):
		text = "NC"
		try:
			temp = requests.get('http://'+self.ip+'/get_temp', timeout=3)
			if temp.status_code == 200:
				soup = BeautifulSoup(temp.text, 'html.parser')
				text = 'Inside: {}\u2103'.format(soup.text.strip())

		except requests.exceptions.RequestException as e:	
			print (e)

		return text		

#'http://192.168.20.100/relay1'

File: HomeNode.java

import net.dongliu.requests.body.*;

public class HomeNode{
  private int    numRelays;
  private String ipAddr;
  private String name;
  
  public HomeNode(String name, String ipAddr, int numRelays){
    this.name      = name;
    this.ipAddr    = ipAddr;
    this.numRelays = numRelays;
  }
  
  public void setRelay(int relay, boolean state){
    if((relay < numRelays) && (relay > 0)){
        requests.get(String.format("http://%s/re%d_%s", ipAddr, relay, state))).send();
    }
  }

  public String getTemp(){
    String temp = "N/A";
    
    return temp;
  }

}
